newIDE/app/src/UI/DragAndDrop/CustomDragLayer.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+import * as React from 'react';
+import { DragLayer } from 'react-dnd';
+import { Identifier } from 'dnd-core';
+import Text from '../Text';
+import { instancesEditorId } from '../../InstancesEditor';
+
+const layerStyles = {
+  position: 'fixed',
+  pointerEvents: 'none',
+  zIndex: 100,
+  left: 0,
+  top: 0,
+  width: '100%',
+  height: '100%',
+};
+
+const THUMBNAIL_SIZE = 48;
+
+const getItemStyles = ({
+  clientOffset,
+  previewPosition,
+}: {|
+  clientOffset: ?{ x: number, y: number },
+  previewPosition: 'center' | 'aboveRight',
+|}) => {
+  if (!clientOffset) {
+    return {
+      display: 'none',
+    };
+  }
+
+  const { x, y } = clientOffset;
+
+  const previewX = previewPosition === 'center' ? x - THUMBNAIL_SIZE / 2 : x;
+  const previewY =
+    previewPosition === 'center'
+      ? y - THUMBNAIL_SIZE / 2
+      : previewPosition === 'aboveRight'
+      ? y - THUMBNAIL_SIZE
+      : y;
+
+  const transform = `translate(${previewX}px, ${previewY}px)`;
+
+  return {
+    transform,
+    WebkitTransform: transform,
+  };
+};
+
+type XYCoord = {|
+  x: number,
+  y: number,
+|};
+
+export type DraggedItem = {|
+  name: string,
+  thumbnail?: string,
+|};
+
+type InternalCustomDragLayerProps = {|
+  item?: DraggedItem,
+  itemType?: Identifier | null,
+  initialOffset?: XYCoord | null,
+  currentOffset?: XYCoord | null,
+  clientOffset?: XYCoord | null,
+  isDragging?: boolean,
+|};
+
+const CustomDragLayer = ({
+  item,
+  itemType,
+  isDragging,
+  initialOffset,
+  currentOffset,
+  clientOffset,
+}: InternalCustomDragLayerProps) => {
+  const renderedItem = React.useMemo(
+    () => {
+      if (!item || !clientOffset) return null;
+
+      const { x, y } = clientOffset;
+      const activeCanvas = document.querySelector(
+        `#scene-editor[data-active=true] #${instancesEditorId}`
+      );
+      if (activeCanvas) {
+        const canvasRect = activeCanvas.getBoundingClientRect();
+        if (
+          x >= canvasRect.left &&
+          x <= canvasRect.right &&
+          y >= canvasRect.top &&
+          y <= canvasRect.bottom
+        ) {
+          return null;
+        }
+      }
+
+      return item.thumbnail ? (
+        <img
+          alt={item.name}
+          src={item.thumbnail}
+          style={{
+            maxWidth: THUMBNAIL_SIZE,
+            maxHeight: THUMBNAIL_SIZE,
+          }}
+        />
+      ) : (
+        <Text>{item.name}</Text>
```

Comment:
I have a feeling that it could give something weird with auto translate, could you check?

Reply:
Good catch, just tried and no issue 👍